hoot-core/src/main/cpp/hoot/core/filters/WaterwayCriterion.cpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2016 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "WaterwayCriterion.h"
+
+namespace hoot
+{
+
+WaterwayCriterion::WaterwayCriterion(FilterType type)
+  : TagContainsFilter(type, "type", "waterway")
```

Comment:
Rather than using the TagContainsFilter it would be better to call OsmSchema::isLinearWaterway. That function is influenced by the schema files (`conf/schema/waterway.json`) and will be updated automagically as more complex waterway types are added. See `HighwayCriterion` for an example.